packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -143,28 +143,138 @@ class FileAccessLog extends AccessLog {
    * @default - no file based access logging
    */
   public readonly filePath: string;
+  private readonly virtualNodeLoggingFormat?: CfnVirtualNode.LoggingFormatProperty;
+  private readonly virtualGatewayLoggingFormat?: CfnVirtualGateway.LoggingFormatProperty;
 
-  constructor(filePath: string) {
+  constructor(filePath: string, loggingFormat?: LoggingFormat) {
     super();
     this.filePath = filePath;
+    // For now we have the same setting for Virtual Gateway and Virtual Nodes
+    this.virtualGatewayLoggingFormat = loggingFormat?.bind().formatConfig;
+    this.virtualNodeLoggingFormat = loggingFormat?.bind().formatConfig;
   }
 
   public bind(_scope: Construct): AccessLogConfig {
     return {
       virtualNodeAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualNodeLoggingFormat,
         },
       },
       virtualGatewayAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualGatewayLoggingFormat,
         },
       },
     };
   }
 }
 
+/**
+ * All Properties for Envoy Access Logging Format for mesh endpoints
+ */
+export interface LoggingFormatConfig {
+  /**
+   * CFN configuration for Access Logging Format
+   *
+   * @default - no access logging format
+   */
+  readonly formatConfig?: CfnVirtualNode.LoggingFormatProperty;
+}
+
+/**
+ * Configuration for Envoy Access Logging Format for mesh endpoints
+ */
+export abstract class LoggingFormat {
+  /**
+   * Generate logging format from text pattern
+   */
+  public static fromText(text: string): LoggingFormat {
+    return new TextLoggingFormat(text);
+  }
+  /**
+   * Generate logging format from json key pairs
+   */
+  public static fromJson(jsonLoggingFormat :{[key:string]: string}): LoggingFormat {
+    const json: JsonFormatRef[] = [];
+    if (Object.keys(jsonLoggingFormat).length == 0) {
+      throw new Error('Json key pairs cannot be empty.');
+    }
+
+    for (const key in jsonLoggingFormat) {
+      json.push(new JsonFormatRef(key, jsonLoggingFormat[key]));
+    };
+
+    return new JsonLoggingFormat(json);
+  };
+
+  /**
+   * Called when the Access Log Format is initialized. Can be used to enforce
+   * mutual exclusivity with future properties
+   */
+  public abstract bind(): LoggingFormatConfig;
+}
+
+/**
+ * Configuration for Json logging format
+ */
+class JsonLoggingFormat extends LoggingFormat {
+  /**
+  * Json pattern for the output logs
+  *
+  * @default - no json format specified
+  */
+  private readonly json?: Array<CfnVirtualNode.JsonFormatRefProperty>;
+  constructor(json: Array<CfnVirtualNode.JsonFormatRefProperty>) {
+    super();
+    this.json = json;
+  }
+
+  bind(): LoggingFormatConfig {
+    return {
+      formatConfig: {
+        json: this.json,
+      },
+    };
+  }
+}
+
+class TextLoggingFormat extends LoggingFormat {
+  /**
+  * Json pattern for the output logs
+  *
+  * @default - no text format specified
+  */
+  private readonly text?: string;
```

Comment:
Why is this optional?

Reply:
The whole field is optional. And you can have either text or json format

Reply:
Sure, but if want the `text` field in the L1 to be undefined, why bother creating an instance of this class in the first place? In fact, you are already enforcing the text to be defined in the public interface (`LoggingFormat.fromText()`). Do the same here. 

Same for the JSON case.

Reply:
Yeah, you are right, being silly here, updated